### ISSUE TITLE
Starting JuputerLab via reverse proxy = blank page

### ISSUE THREAD

**Issue by haf** [jupyterlab/jupyterlab#11393]
## Description

Putting JupyterLab behinda reverse proxy yields only a blank screen.

## Reproduce

1. Deploy via https://github.com/dask/helm-chart on k8s
2. Reverse proxy it
3. White page

But:

1. same 1
2. `k port-forward svc/dask-jupyter 8090:80`
3. Works

## Expected behavior

Should show output. Now it shows this

<img width="1425" alt="image" src="https://user-images.githubusercontent.com/193115/139875842-cf64bfe4-570e-429b-b2fe-99188eef19b8.png">

## Context

Tried in Chromium, Safari and FF evergreen.

<img width="401" alt="image" src="https://user-images.githubusercontent.com/193115/139875949-9e6b3aeb-cf90-4718-9d7d-39ed907edaea.png">

Reverse proxy is envoy / istio. Works for all other apps.

**Comment by krassowski** [comment]
Welcome!

1. I'm a bit lost here - the screenshot shows all successful requests, but presumably some of those failed as you mention "permission denied" in the other repository. Could it be that you need to scroll down/up to capture the relevant part? Or are they all indeed successful requests?
2. Are there any logs in the 'Console' tab of developer tools?
3. It might be that it has something to do with JupyterHub configuration (hard to tell right now, but I guess it is more likely than a bug in JupyterLab itself), and it might be that asking on JupyterHub gitter or Jupyter Discourse might be of more help to you.

**Comment by haf** [comment]
@krassowski Thank you :)

1. They are actually all successful
   - the permission denied in the other repository had permission denied *in the UI when port-forwarding also*, so they are separate issues and the permission errors were resolved by not setting a working directory in the dockerfile
2. zero logs outputted, no errors on the server logs
3. I'm not using JupyterHub, I tried that, it randomly stopped working; now I've downgraded to only one instance, so I don't see how that would take me forward

Is there a verbose logging flag I can set?

**Comment by krassowski** [comment]
`jupyter lab --debug`? Are you using `https`/configured your own certificate?

**Comment by haf** [comment]
Second question: yes

**Comment by krassowski** [comment]
I have seen some browsers render blank pages when they encounter an untrusted certificate (as in: not in the browser CA store). A solution is to close the tab and open it again, confirming that you trust the certificate. Could it be related, or are you seeing the blank page consistently?

**Comment by haf** [comment]
No, the certificate is a Let's Encrypt certificate and it's valid.

The blank page comes consistently when reverse-proxying the notebook and the problem for me operationally is that I don't see why; the logs show nothing. (I haven't tried --debug yet)

**Comment by haf** [comment]
Were you able to reproduce it :)?

**Comment by haf** [comment]
Ping

**Comment by haf** [comment]
Ping
